### ISSUE TITLE
import/export SMS & MMS

### ISSUE THREAD

**Issue by FilLupin** [SilenceIM/Silence#597]
Hi there,
I am interested into understanding how to import and export SMS and MMS.
This is more a summary of current situatin than a bug but it seems useful to me since I did not find anything in the doc.
Sorry of there is anything already existing, could you then indicate me where I can find it?

*Import SMS & MMS when installing Silence for the first time:* if I understand well (especially #140 and #184 ), there is no way to import external MMS into Silence.

*Silence SMS & MMS export* (or just SMS when not encrypted export):
- it is available only inside the app (unavailable if crash at starting like in #594 )
- it only include SMS and possibly MMS (if encrypted export) but not exchanged keys if I understand well #382 )
- its format is not compatible with every Silence version (as written in #594 ). A table with full compatibility seems a must to me. An import/export from/to several version would be very useful to allow newcomers to adopt Silence.

*Manual SMS & MMS export/backup:* 
- one can manually backup SMS from command line with adb root access with command:
> adb pull /data/data/org.smssecure.smssecure

But I do not understand what this backup will include?
And so I do not know what I will get when getting back the backup with 
> adb root
> adb push ./SAVE_DIR /data/data/org.smssecure.smssecure

This would be very usefull to allow to not loose data if Silence crashes (like #594 ).

I will perhaps complete this post if I find more questions/answers.

**Comment by ArchangeGabriel** [comment]
You’re right on everything you’ve stated (now that you have edited your post), and now regarding your questions:

- Manual backup: I’m not sure what `adb pull` will include. However note that `adb backup` does not work, because Silence protects its data from being accessed. Maybe `adb pull` as root work around this, I suppose so but would need to try.
- Uninstalling means everything is deleted. Upon reinstall, a restore is necessary to recover data, else Silence will just be empty.
- There is indeed actually no way to export MMS out of Silence, excepted for Silence backups (which only Silence knows how to use). Especially, you cannot move them to another app.

Solving #140/#184 is required to remove most of your concerns. Regarding backup without Silence working, the only thing you can do is having backups before this happen. Backups are not the thing you want to do once an issue appears. They must be done beforehand. At the very best a script could be made to extract data from Silence DB using recovery or root.

**Comment by FilLupin** [comment]
Yes, I completed my posts two times to add questions and answers or precisions. :)
Thank you for your answers.

- manual backup : I would be glad to make some tests, but I do not have actually a phone I can use (I do not have full backup through encrypted export). So if someone with a Silence installation without bug can make some tests, it would help. Especially, I would like to check MMS are included into this backup.
- uninstalling : this make sens, but I wanted a confirmation, thanks !
- MMS export out of Silence : very bad news, do you know if this is due to some specific difficulty or is that a development choice?

I understand well my situation here, and installing unstable version was of course an error. In fact, I installed silence in order to convert old SMS from an android 4 to an android 6 (it was the only solution I found). Now, I am stuck with Silence so I would like to find a solution without loosing every MMS.
I made some backups but since Silence can not program an automatic backup, I did not made a daily backup and here am I.
If you think of any solution, it would be very useful.

**Comment by ArchangeGabriel** [comment]
MMS export not working is a bug, not a choice. ;)

I’ve made the same mistake that you did (upgraded to unstable Silence, which was broken, and then lost 4 months of messages by uninstalling because I didn’t have backups more recent than that).

About LibreSignal, does this one even support SMS? I think the fork happened after SMS support removal. And even so, I have absolutely no idea whether the format is compatible. @BLeQuerrec and @pR0Ps are the one that might know about this, I’m just an user.

**Comment by FilLupin** [comment]
OK, so you're right since LibreSignal is not a fork but just an attempt to make build reproduicable. It uses the same code as Signal, the slight difference is that each commit is checked to assure no backdoor is included.

**Comment by FilLupin** [comment]
Hello, I tried manual backup, uninstalling then reinstalling and copy back backuped directory, but as said in #594 it failed. Do you have an idea how to do this?
Thank you

**Comment by FilLupin** [comment]
up, in case anyone has any clue

**Comment by FilLupin** [comment]
I've seen the news in #610 and #612 which seem a very good news for Silence.
I come back to check if the author would have any clue how to solve this situation I encountered here?
